### PR TITLE
`CompareKeccakFullRateInBytes` keccak hint 

### LINKED ISSUE #480: `compareKeccakFullRateInBytes` hint (Keccak hint)
following hint is missing:

https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/cairo_keccak/keccak.cairo#L310

```
ids.n_bytes >= ids.KECCAK_FULL_RATE_IN_BYTES
```

it is used in `_keccak` function that executes the block permutation

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #480 

Implementation is from @MaksymMalicki  in #462 , just adapted hintcode and left only the hint

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 1]
```diff
@@ -506,7 +506,7 @@ func newSquashDictHint(dictAccesses, ptrDiff, nAccesses, bigKeys, firstKey hinte
 			//  __squash_dict_max_size is always in scope and has a value of 2**20,
 			squashDictMaxSize := uint64(1048576)
 			if nAccessesValue > squashDictMaxSize {
-				return fmt.Errorf("squash_dict() can only be used with n_accesses<={%d}. Got: n_accesses={%d}.", squashDictMaxSize, nAccessesValue)
+				return fmt.Errorf("squash_dict() can only be used with n_accesses<={%d}. Got: n_accesses={%d}", squashDictMaxSize, nAccessesValue)
```
Comment (Sh0g0-1758) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:509]: If I remember correctly, I think there were many of these formatting issues with go in the code base. Maybe we can do them in this PR or another PR ?

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 2]
```diff
@@ -407,6 +407,45 @@ func createKeccakWriteArgsHinter(resolver hintReferenceResolver) (hinter.Hinter,
...
+//   - `nBytes` is the value to be compared with KECCAK_FULL_RATE_IN_BYTES
+//
+// `newKeccakWriteArgsHint` writes 1 or 0 to `ap` memory address depending on whether
+// `n_bytes` is greater or equal to KECCAK_FULL_RATE_IN_BYTES or not
+func newCompareKeccakFullRateInBytesHint(nBytes hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "CompareKeccakFullRateInBytes",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> ids.n_bytes >= ids.KECCAK_FULL_RATE_IN_BYTES
+			nBytesVal, err := hinter.ResolveAsUint64(vm, nBytes)
```
Comment (Sh0g0-1758) [inline pkg/hintrunner/zero/zerohint_keccak.go:422]: Same as previous hint. Should we not resolve this as a felt ? 

From cairo-lang : `n_bytes: felt`

File: pkg/hintrunner/zero/zerohint_dictionaries.go (hunk 1 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_dictionaries.go:509]: I'm not sure there are still after these one in zero package
If you find some open another one if you want

File: pkg/hintrunner/zero/zerohint_keccak.go (hunk 2 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak.go:422]: mmh,
n_bytes is:

```
n_bytes=n_elements * 32
```

n_elements is a felt, but represents the number of uint256  (or felts) on which we apply the keccak.

It seems logic that n_elements will never be bigger than uint64, but we never know?  and `n_bytes` is `n_elements * 32` , so i guess theoretically we should use felts indeed here?

@cicr99  what do you think?

File: pkg/hintrunner/zero/zerohint_dictionaries.go (hunk 1 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_dictionaries.go:509]: okay.

File: pkg/hintrunner/zero/hintcode.go [hunk 3]
```diff
@@ -125,8 +125,9 @@ const (
 	inp = [0] * _keccak_state_size_felts
 	padding = (inp + keccak_func(inp)) * _block_size
 	segments.write_arg(ids.keccak_ptr_end, padding)`
-	keccakWriteArgsCode  string = "segments.write_arg(ids.inputs, [ids.low % 2 ** 64, ids.low // 2 ** 64])\nsegments.write_arg(ids.inputs + 2, [ids.high % 2 ** 64, ids.high // 2 ** 64])"
-	blockPermutationCode string = "from starkware.cairo.common.keccak_utils.keccak_utils import keccak_func\n_keccak_state_size_felts = int(ids.KECCAK_STATE_SIZE_FELTS)\nassert 0 <= _keccak_state_size_felts < 100\noutput_values = keccak_func(memory.get_range(\nids.keccak_ptr - _keccak_state_size_felts, _keccak_state_size_felts))\nsegments.write_arg(ids.keccak_ptr, output_values)"
+	keccakWriteArgsCode              string = "segments.write_arg(ids.inputs, [ids.low % 2 ** 64, ids.low // 2 ** 64])\nsegments.write_arg(ids.inputs + 2, [ids.high % 2 ** 64, ids.high // 2 ** 64])"
+	compareKeccakFullRateInBytesCode string = "ids.n_bytes >= ids.KECCAK_FULL_RATE_IN_BYTES"
```
Comment (TAdev0) [inline pkg/hintrunner/zero/hintcode.go:129]: seems that hintcode should be:

"memory[ap] = to_felt_or_relocatable(ids.n_bytes >= ids.KECCAK_FULL_RATE_IN_BYTES)"

File: pkg/hintrunner/zero/zerohint_keccak.go (hunk 2 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak.go:422]: ok will add a comment, thanks @cicr99  !

**Comment by TAdev0** [comment]
@cicr99  added a comment, should be good!
